src/wp-includes/block-template-utils.php

```diff
@@ -507,6 +507,64 @@ function _remove_theme_attribute_from_template_part_block( &$block ) {
 	}
 }
 
+/**
+ * Gets the contents for the given template file path, using a transient cache.
+ *
+ * @since 6.4.0
+ * @access private
+ *
+ * @param string $template_file_path Absolute path to a theme template file.
+ * @return string The template file contents.
+ */
+function _get_block_template_file_content( $template_file_path ) {
+	$theme = wp_get_theme();
+
+	$template_file_path = wp_normalize_path( $template_file_path );
+	$theme_dir          = wp_normalize_path( $theme->get_stylesheet_directory() );
+
+	if ( str_starts_with( $template_file_path, $theme_dir ) ) {
+		$relative_path = substr( $template_file_path, strlen( $theme_dir ) );
+	} elseif ( $theme->parent() ) {
+		$theme     = $theme->parent();
+		$theme_dir = wp_normalize_path( $theme->get_stylesheet_directory() );
+		if ( str_starts_with( $template_file_path, $theme_dir ) ) {
+			$relative_path = substr( $template_file_path, strlen( $theme_dir ) );
+		}
+	}
+
+	// Bypass cache if the file is not within the theme directory.
+	if ( ! isset( $relative_path ) ) {
+		return file_get_contents( $template_file_path );
+	}
+
+	// Bypass cache while developing a theme.
+	$can_use_cache = ! wp_is_development_mode( 'theme' );
+	if ( ! $can_use_cache ) {
+		return file_get_contents( $template_file_path );
+	}
```

Comment:
I'd combine these two if statements and run the caching logic below only if both conditions are truthy. Otherwise, `return file_get_contents();` as the final step in the function.

Ex:

```php
if ( isset( $relateive_path) && $can_use_cache ) {
  // transient logic goes here
  return $template_data['template_content'][ $relative_path ];
}

return file_get_contents( $template_file_path );
```

Reply:
By doing that, we'll have to check `$can_use_cache` twice, so I think early returns for not using the cache are more efficient.